nntrainer/graph/network_graph.cpp

```diff
@@ -268,33 +267,44 @@ int NetworkGraph::realizeActivationType(Layer &current) {
   current.output_layers.clear();
   current.output_layers.push_back(layer->getName());
 
-  addLayerNode(layer);
-
   updateNameInLayers(current.getName(), layer->getName());
+  addLayerNode(layer);
 
   return status;
 }
 
 int NetworkGraph::addLossLayer(const LossType loss_type) {
+
   int status = ML_ERROR_NONE;
+
```

Comment:
Check if Sorted.empty() ?

Reply:
this is already guaranteed with `isCompilable()` which ensures that `adj` is not empty. And `topologicalSort` by definition ensures that Sorted will not be empty. 
For completeness, added a check at the end of topologicalSort.

Reply:
Cool